source/Cosmos.System2_Plugs/System/Runtime/CompilerServicesImpl.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Cosmos.IL2CPU.API.Attribs;
+
+namespace Cosmos.System_Plugs.System.Runtime
+{
+    [Plug(Target = typeof(global::System.Runtime.CompilerServices.RuntimeHelpers))]
+    public static class RuntimeHelpersImpl
+    {
+        /* For RAT in Cosmos the size of Reference Types is 8 on x86 / 32 bit too...
+         * String object header has two reference types so the data part is at offset 16
```

Comment:
This comment is not totally correct.
1. References are: `(currentAddress << sizeof(IntPtr)) | nextAddress`, where `currentAddress` and `nextAddress` are `IntPtr`.
2. The data part starts at offset 12, but the first char is at offset 16, which is what this property returns. From the comment in CoreCLR (notice that it skips over the string length, which is the first 4 bytes after the object header):
>// Number of bytes from the address pointed to by a reference to
// a String to the first 16-bit character in the String.  Skip 
// over the MethodTable pointer, & String 
// length.